bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/BLockStore.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *   Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.ballerinalang.jvm;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ *
+ * @since 1.2.0
+ */
+public class BLockStore {
+
+    /**
+     * The map of locks inferred.
+     */
+    private  Map<String, BLock> globalLockMap;
+
+    public BLockStore() {
+        globalLockMap = new ConcurrentHashMap<>();
```

Comment:
Shouldn't this be a thread sage singleton

Reply:
This class is accessed only once per program, using the init class in jbal